exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/handlers/DescribeSchemaHandler.java

```diff
@@ -0,0 +1,129 @@
+ * <p/>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p/>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.drill.exec.planner.sql.handlers;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.SerializableString;
+import com.fasterxml.jackson.core.io.CharacterEscapes;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Joiner;
+import org.apache.calcite.schema.SchemaPlus;
+import org.apache.calcite.sql.SqlIdentifier;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.drill.common.exceptions.ExecutionSetupException;
+import org.apache.drill.common.exceptions.UserException;
+import org.apache.drill.common.logical.StoragePluginConfig;
+import org.apache.drill.exec.physical.PhysicalPlan;
+import org.apache.drill.exec.planner.sql.DirectPlan;
+import org.apache.drill.exec.planner.sql.SchemaUtilites;
+import org.apache.drill.exec.planner.sql.parser.SqlDescribeSchema;
+import org.apache.drill.exec.store.StoragePlugin;
+import org.apache.drill.exec.store.dfs.FileSystemSchemaFactory;
+import org.apache.drill.exec.store.dfs.WorkspaceConfig;
+
+import java.util.List;
+import java.util.Map;
+
+import static com.fasterxml.jackson.databind.SerializationFeature.INDENT_OUTPUT;
+
+public class DescribeSchemaHandler extends DefaultSqlHandler {
+
+  public DescribeSchemaHandler(SqlHandlerConfig config) {
+    super(config);
+  }
+
+  private static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(DescribeSchemaHandler.class);
+  private static final ObjectMapper mapper = new ObjectMapper(new ObjectMapper().getFactory().setCharacterEscapes(new CharacterEscapes() {
+    @Override
+    public int[] getEscapeCodesForAscii() {
+      // add standard set of escaping characters
+      int[] esc = CharacterEscapes.standardAsciiEscapesForJSON();
+      // don't escape backslash (not to corrupt windows path)
+      esc['\\'] = CharacterEscapes.ESCAPE_NONE;
+      return esc;
+    }
+
+    @Override
+    public SerializableString getEscapeSequence(int i) {
+      // no further escaping (beyond ASCII chars) needed
+      return null;
+    }
+  })).enable(INDENT_OUTPUT);
+
+
+  @Override
+  public PhysicalPlan getPlan(SqlNode sqlNode) {
+    SqlIdentifier schema = ((SqlDescribeSchema) sqlNode).getSchema();
+    SchemaPlus drillSchema = SchemaUtilites.findSchema(config.getConverter().getDefaultSchema(), schema.names);
+
+    if (drillSchema != null) {
+      StoragePlugin storagePlugin;
+      try {
+        storagePlugin = context.getStorage().getPlugin(schema.names.get(0));
+      } catch (ExecutionSetupException e) {
+        throw UserException.validationError()
```

Comment:
No particular reason. replaced it with DrillRuntimeException (searched through Drill code, mostly DrillRuntimeException is re-thrown instead of ExecutionSetupexception).